src/events/api/serializers.py

```diff
@@ -11,117 +11,168 @@ class PrimarySpeakerSerializer(serializers.Serializer):
+    flattening the `proposal` object from serialized
+    `ProposedTalkEvent` or `ProposedTutorialEvent`.
+    """
+    proposal_repr = representation.pop('proposal')
+    for key in proposal_repr:
+        representation[key] = proposal_repr[key]
+    return representation
+
+
+class TalkProposalSerializer(serializers.ModelSerializer):
     speakers = serializers.SerializerMethodField()
+    is_sponsored = serializers.ReadOnlyField(default=False)
 
     def get_speakers(self, obj):
-        return [
-            ReturnDict(PrimarySpeakerSerializer(
-                data={'thumbnail_url': i.user.get_thumbnail_url(),
-                      'name': i.user.speaker_name,
-                      'github_profile_url': i.user.github_profile_url,
-                      'twitter_profile_url': i.user.twitter_profile_url,
-                      'facebook_profile_url': i.user.facebook_profile_url}).get_initial(),
-                serializer=PrimarySpeakerSerializer) for i in obj.speakers]
+        request = self.context.get('request')
+        users = [s.user for s in obj.speakers]
+        return format_speakers_data(request, users, show_details=True)
 
     class Meta:
         model = TalkProposal
         fields = [
-            "title",
-            "category",
-            "language",
-            "python_level",
-            "recording_policy",
-            "abstract",
-            "detailed_description",
-            "slide_link",
-            "slido_embed_link",
-            # "sponsored"
-            "speakers"
-
+            "title", "category", "language", "python_level",
+            "recording_policy", "abstract", "detailed_description",
+            "slide_link", "slido_embed_link", "speakers",
+            "is_sponsored",
         ]
 
 
+class TalkDetailSerializer(serializers.ModelSerializer):
+    proposal = TalkProposalSerializer()
+
+    def to_representation(self, obj):
+        representation = super().to_representation(obj)
+        return flatten_proposal_field(representation)
+
+    class Meta:
+        model = ProposedTalkEvent
+        fields = ['proposal', 'begin_time', 'end_time', 'is_remote', 'location']
+
+
 class TalkListSerializer(serializers.ModelSerializer):
     speakers = serializers.SerializerMethodField()
+    is_sponsored = serializers.ReadOnlyField(default=False)
 
     def get_speakers(self, obj):
-        return [
-            ReturnDict(PrimarySpeakerSerializer(
-                data={'thumbnail_url': i.user.get_thumbnail_url(),
-                      'name': i.user.speaker_name,
-                      'github_profile_url': i.user.github_profile_url,
-                      'twitter_profile_url': i.user.twitter_profile_url,
-                      'facebook_profile_url': i.user.facebook_profile_url}).get_initial(),
-                serializer=PrimarySpeakerSerializer) for i in obj.speakers]
+        request = self.context.get('request')
+        users = [s.user for s in obj.speakers]
```

Comment:
I think it is more readable to replace the parameter name from users to speakers.